<launch>
    <node name="husky_rqt_dashboard" pkg="rqt_gui" type="rqt_gui" respawn="false" output="screen" args="--perspective-file $(find kimm_multi_husky_gui)/config/gui_rviz.perspective" />
	<arg name="rvizconfig" default="$(find kimm_multi_husky_gui)/config/husky_rviz.rviz" />

	<param name="robot_group0" type="string" value="ns0" />
	<param name="robot_group1" type="string" value="ns1" />
	
	<group ns ="ns0_gui">
	    <param name="robot_description" command="cat $(find husky_description)/husky_single/husky_panda_rviz_ns0.urdf" />
		<node pkg="tf2_ros" type="static_transform_publisher" name="odom" args="0 0 0 0 0 0 world panda_link0" />
		<param name="rviz_urdf_path" type="string" value="$(find husky_description)"/>
   		<param name="rviz_urdf" type="string" value="/husky_single/husky_panda_rviz_ns0.urdf"/>
		<node name="robot_state_publisher" pkg="robot_state_publisher"
			type="robot_state_publisher" respawn="true">
			<param name="publish_frequency" value="100" />
			<param name="rate" value="100" />
		</node>
		<node name="kimm_joint_planner_ros_interface_server" pkg="kimm_joint_planner_ros_interface" type="kimm_joint_planner_ros_interface_server" output="screen" >
		</node>
		<node name="kimm_se3_planner_ros_interface_server" pkg="kimm_se3_planner_ros_interface" type="kimm_se3_planner_ros_interface_server" output="screen" >
		</node>
        <node name="kimm_path_planner_ros_interface_server" pkg="kimm_path_planner_ros_interface" type="kimm_path_planner_ros_interface_server" output="screen" >
            <param name="save_path" type="string" value="$(find kimm_path_planner_ros_interface)/result/mobile.json"/>    
            <param name="robot_planning_model" type="string" value="$(find kimm_path_planner)/robot_model/car.svg"/>
            <param name="plan_srv_name" type="string" value="plan_mobile_path"/>
            <param name="action_srv_name" type="string" value="action_mobile_path"/>
		</node>
	</group>
	<group ns ="ns1_gui">
	    <param name="robot_description" command="cat $(find husky_description)/husky_single/husky_panda_rviz_ns1.urdf" />
		<node pkg="tf2_ros" type="static_transform_publisher" name="odom" args="0 0 0 0 0 0 world panda_link0" />
		<param name="rviz_urdf_path" type="string" value="$(find husky_description)"/>
   		<param name="rviz_urdf" type="string" value="/husky_single/husky_panda_rviz_ns1.urdf"/>
		<node name="robot_state_publisher" pkg="robot_state_publisher"
			type="robot_state_publisher" respawn="true">
			<param name="publish_frequency" value="100" />
			<param name="rate" value="100" />
		</node>
		<node name="kimm_joint_planner_ros_interface_server" pkg="kimm_joint_planner_ros_interface" type="kimm_joint_planner_ros_interface_server" output="screen" >
		</node>
		<node name="kimm_se3_planner_ros_interface_server" pkg="kimm_se3_planner_ros_interface" type="kimm_se3_planner_ros_interface_server" output="screen" >
		</node>
        <node name="kimm_path_planner_ros_interface_server" pkg="kimm_path_planner_ros_interface" type="kimm_path_planner_ros_interface_server" output="screen" >
            <param name="save_path" type="string" value="$(find kimm_path_planner_ros_interface)/result/mobile.json"/>    
            <param name="robot_planning_model" type="string" value="$(find kimm_path_planner)/robot_model/car.svg"/>
            <param name="plan_srv_name" type="string" value="plan_mobile_path"/>
            <param name="action_srv_name" type="string" value="action_mobile_path"/>
		</node>
	</group>
</launch>
